tests/Convert-SentinelARArmToYaml.tests.ps1

```diff
@@ -104,115 +267,310 @@ Describe "Convert-SentinelARArmToYaml" {
+        }
+        It "Pipeline and OutFile" {
+            Get-Content -Path "TestDrive:/Single/Scheduled.json" -Raw | Convert-SentinelARArmToYaml -OutFile "TestDrive:/Single/Scheduled.yaml"
+            Get-ChildItem -Path "TestDrive:/Single/*" -Include *.yaml | Should -HaveCount 1
+        }
+    }
+    Context "Multiple example tests" -Tag Integration {
+        BeforeAll {
+            New-Item TestDrive:/Multiple/ -ItemType Directory | Out-Null
+            Copy-Item -Path $exampleMultipleFilePath -Destination TestDrive:/Multiple/
+        }
+        AfterEach {
+            Remove-Item -Path "TestDrive:/Multiple/*" -Include *.yaml -Force
+        }
+        It "No Pipeline and Outfile" {
+            Convert-SentinelARArmToYaml -Filename "TestDrive:/Multiple/ScheduledMultiple.json" -OutFile "TestDrive:/Multiple/ScheduledMultiple.yaml"
+            Get-ChildItem -Path "TestDrive:/Multiple/*" -Include *.yaml | Should -HaveCount 2
+        }
+        It "No Pipeline and UseOriginalFilename" {
+            Convert-SentinelARArmToYaml -Filename "TestDrive:/Multiple/ScheduledMultiple.json" -UseOriginalFilename
+            Get-ChildItem -Path "TestDrive:/Multiple/*" -Include *.yaml | Should -HaveCount 2
+        }
+        It "No Pipeline and UseDisplayNameAsFilename" {
+            Convert-SentinelARArmToYaml -Filename "TestDrive:/Multiple/ScheduledMultiple.json" -UseDisplayNameAsFilename
+            Get-ChildItem -Path "TestDrive:/Multiple/*" -Include *.yaml | Should -HaveCount 2
+        }
+        It "No Pipeline and UseIdAsFilename" {
+            Convert-SentinelARArmToYaml -Filename "TestDrive:/Multiple/ScheduledMultiple.json" -UseIdAsFilename
+            Get-ChildItem -Path "TestDrive:/Multiple/*" -Include *.yaml | Should -HaveCount 2
+        }
+        It "Pipeline and OutFile" {
+            Get-Content -Path "TestDrive:/Multiple/ScheduledMultiple.json" -Raw | Convert-SentinelARArmToYaml -OutFile "TestDrive:/Multiple/ScheduledMultiple.yaml"
+            Get-ChildItem -Path "TestDrive:/Multiple/*" -Include *.yaml | Should -HaveCount 2
+        }
+        It "Pipeline and UseDisplayNameAsFilename" {
+            Get-Content -Path "TestDrive:/Multiple/ScheduledMultiple.json" -Raw | Convert-SentinelARArmToYaml -UseDisplayNameAsFilename -Directory "TestDrive:/Multiple/"
+            Get-ChildItem -Path "TestDrive:/Multiple/*" -Include *.yaml | Should -HaveCount 2
+        }
+        It "Pipeline and UseIdAsFilename" {
+            Get-Content -Path "TestDrive:/Multiple/ScheduledMultiple.json" -Raw | Convert-SentinelARArmToYaml -UseIdAsFilename -Directory "TestDrive:/Multiple/"
+            Get-ChildItem -Path "TestDrive:/Multiple/*" -Include *.yaml | Should -HaveCount 2
         }
     }
 }
 
 AfterAll {
-    Remove-Item -Path "$PSScriptRoot\testOutput" -Recurse -Force
+    if (-not $RetainTestFiles) {
+        Remove-Item -Path "$PSScriptRoot/testOutput/" -Recurse -Force
+    }
```

Comment:
```suggestion
    if (-not $RetainTestFiles) {
        Remove-Item -Path "$PSScriptRoot/testOutput/" -Recurse -Force
    }
    if ( Get-Module -Name SentinelARConverter ) {
        Remove-Module -Name SentinelARConverter -Force
    }
```

Reply:
To avoid issues with the already loaded module let us remove it at the end of the tests